src/DDTrace/Integrations/ZendFramework/ZendFrameworkSandboxedIntegration.php

```diff
@@ -0,0 +1,132 @@
+    {
+        return self::NAME;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function requiresExplicitTraceAnalyticsEnabling()
+    {
+        return false;
+    }
+
+    /**
+     * Loads the zend framework integration.
+     *
+     * @return int
+     */
+    public function init()
+    {
+        if (!self::shouldLoad(self::NAME)) {
+            return self::NOT_AVAILABLE;
+        }
+
+        // Some frameworks, e.g. Yii registers autoloaders that fails with non-psr4 classes. For this reason the
+        // Zend framework integration is not compatible with some of them
+        if (Runtime::isAutoloaderRegistered('YiiBase', 'autoload')) {
+            return self::NOT_AVAILABLE;
+        }
+
+        $integration = $this;
+        // For backward compatibility with the legacy API we are not using the integration
+        // name 'zendframework', we are instead using the 'zf1' prefix.
+        $appName = Configuration::get()->appName('zf1');
+
+        dd_trace_method(
+            'Zend_Controller_Plugin_Broker',
+            'preDispatch',
+            function (SpanData $spanData, $args) use ($integration, $appName) {
+                $rootScope = GlobalTracer::get()->getRootScope();
+                if (null === $rootScope) {
+                    return false;
+                }
+
+                $rootSpan = $rootScope->getSpan();
+                if (null === $rootSpan) {
+                    return false;
+                }
+
+                // We are enclosing this into a try-catch because we always have to return false,
+                // even at the cost of not setting specific metadata.
+                try {
+                    /** @var Zend_Controller_Request_Abstract $request */
+                    list($request) = $args;
+                    $rootSpan->setIntegration($integration);
+                    $rootSpan->setTraceAnalyticsCandidate();
+                    $rootSpan->overwriteOperationName($integration->getOperationName());
+                    $rootSpan->setTag(Tag::SERVICE_NAME, $appName);
+                    $controller = $request->getControllerName();
+                    $action = $request->getActionName();
+                    $route = Zend_Controller_Front::getInstance()->getRouter()->getCurrentRouteName();
+                    $rootSpan->setTag('zf1.controller', $controller);
+                    $rootSpan->setTag('zf1.action', $action);
+                    $rootSpan->setTag('zf1.route_name', $route);
+                    $rootSpan->setResource($controller . '@' . $action . ' ' . $route);
+                    $rootSpan->setTag(Tag::HTTP_METHOD, $request->getMethod());
+                    $rootSpan->setTag(
+                        Tag::HTTP_URL,
+                        $request->getScheme() . '://' .
+                        $request->getHttpHost() .
+                        $request->getRequestUri()
+                    );
+                } catch (\Exception $e) {
+                }
```

Comment:
Do you know of any exceptions that might be thrown here?
@SammyK, if an exception gets thrown then the span gets dropped, right? We shouldn't have to do a try-catch in this instance?

Reply:
The problem with this is that if an exception is thrown we do not return `false`. Hence an almost empty span is generated, not having basic info including name/service/resource.

This error would cause the entire trace to be dropped.

Reply:
Do you think it might make sense to add a field like `->transient` to a `SpanData` object that can be set as the first thing and causes the span to be dropped in place of the return false approach? cc @morrisonlevi @SammyK

Reply:
Dropping the span from the tracing closure it problematic since it can [create orphan spans](https://github.com/DataDog/dd-trace-php/pull/608). For hooks like this, we need a separate API that just does the pre or post hooking without any span creation. I created a card for this. :)